airmencoders/Logging-iOS
# PBS-1453 Fix popover oddities

[PBS-1453](https://jira.il2.dso.mil/browse/PBS-1453)

**Discussion** 🗣
This PR addresses a few issues regarding the popovers used throughout the app. The main glitch being addressed occurs when a user tries to open a popover when one is already open. The intended popover will not show, and the popover that is already open will remain open. 

Upon doing some [research](https://developer.apple.com/design/human-interface-guidelines/ios/views/popovers/), I found that Apple strongly recommends that you only have one open at once. If you do need open a second popover, you must make sure the first one has been dismissed first. 

**Important Notes** ✍️
- Popover text no longer flashes black/white (depending on light/dark mode) when dismissing the popover
- Logic has been added to make sure that only one popover is open at a time. If no popover is already open, the intended popover is shown instantaneously. If one is already open, the intended popover will be shown after a brief delay, giving the other popover time to dismiss. 
- A longer delay will occur if the app is run on a simulator, to account for lag.
- If no delay is used, the original issue persists. The previous popover has to be completely gone from the UI before opening the next one.

**Testing** 
- Navigate to the Fuel, Airland, and Airdrop section in Mission Data. Input a very large number for any field pertaining to weight (KLBS), and try and tap on the different error buttons both slowly and rapidly. 
- Navigate to FlightTime view and alternate tapping the Gonkulate/Night Hours info buttons, both slowly and rapidly. 
- In both views, make sure there is only one popover open at a time, and there are no glitches occurring with the text (flashing, displaying the wrong text, etc.).
- Make sure this works in simulator and on device.
